### DISCUSSION TITLE
画面ロングタップ時のCell内への画像表示

### DISCUSSION THREAD

**Discussion by mahirooo** [lit-iphone-course/lit-iphone-course#47] [Question-original] [answered]
(1) 実現したいこと
PushViewControllerをロングタップした時に、"**一個目のCellから**"画像を表示したい

(2) 実現するために自分が試した内容とその結果
現状：　var stampImage = [""]→　ロングタップするごとに画像が表示される(これは◎)。ただ、2個目のCellから表示されてしまう。
試したやつ:　var stampImage = [String] → エラー文「Expected member name or constructor call after type name」

(3) 調べた内容や資料
調べた限り参考になりそうなものがなく.......

(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
`var stampImage = [""]`
[Gitコードリンク](https://github.com/mahirooo/TampStamp/blob/ae55e6914aa40441747677469d0e8b33505d1d46/TampStamp/PushViewController.swift#L13)
```

func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return stampImage.count
    }
    
    internal func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell:UICollectionViewCell = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath)
        let imageView = cell.contentView.viewWithTag(2) as! UIImageView
        let cellImage = UIImage(named: stampImage[indexPath.row])
                // UIImageをUIImageViewのimageとして設定
                imageView.image = cellImage
                return cell
    }
```
[Gitコードリンク](https://github.com/mahirooo/TampStamp/blob/ae55e6914aa40441747677469d0e8b33505d1d46/TampStamp/PushViewController.swift#L61-L72)

**Comment by txxxxc** [comment]
Swiftにおける空の配列の定義の仕方は２つあります！
```
let array = [String]()
let array: [String] = []
```
どっちでも意味は一緒なので試してみてくださいー！
多分これで正しくスタンプ押せるはず！

**Reply by mahirooo** [reply]
上記で配列を定義してみたところ、
1回目長押し⇨3つ目にスタンプが押される
2回目長押し⇨1つめと2つめにスタンプが押される(3つ目には押されていない状態になる)
3回目長押し⇨1つめと2つめと３つ目にスタンプが押される
....

というふうになってしまいました。。。。

Git最新版にPushしてあるので見ていただけると幸いです！

**Comment by txxxxc** [comment, marked as answer]
CollectionViewの`Estimate Size`を`None`にしたら治ります！
一回試してみてくださいー！
![image](https://user-images.githubusercontent.com/22112440/192128469-f79e6719-4997-4087-a5a7-72a747223245.png)

**Reply by mahirooo** [reply]
直りましたー！ありがとうございます！！！
